dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/DefaultCustomIconService.java

```diff
@@ -55,170 +48,172 @@
  * @author Kristian Wærstad
  */
 @RequiredArgsConstructor
-@Service("org.hisp.dhis.icon.IconService")
-public class DefaultIconService implements IconService {
+@Service("org.hisp.dhis.icon.CustomIconService")
+public class DefaultCustomIconService implements CustomIconService {
+
+  private static final String CUSTOM_ICON_KEY_PATTERN = "^[a-zA-Z0-9*-_+]+$";
+
   private static final String ICON_PATH = "SVGs";
 
   private final CustomIconStore customIconStore;
 
   private final FileResourceService fileResourceService;
 
-  private final Map<String, DefaultIcon> defaultIcons =
-      Arrays.stream(DefaultIcon.Icons.values())
-          .map(DefaultIcon.Icons::getVariants)
-          .flatMap(Collection::stream)
-          .collect(Collectors.toMap(DefaultIcon::getKey, Function.identity()));
-
   @Override
   @Transactional(readOnly = true)
-  public List<Icon> getIcons(IconOperationParams iconOperationParams) {
-    if (IconTypeFilter.DEFAULT == iconOperationParams.getIconTypeFilter()) {
-      return defaultIcons.values().stream()
-          .filter(icon -> Set.of(icon.getKeywords()).containsAll(iconOperationParams.getKeywords()))
-          .collect(Collectors.toList());
-    }
-
-    if (IconTypeFilter.CUSTOM == iconOperationParams.getIconTypeFilter()) {
-      return customIconStore.getIcons(iconOperationParams).collect(Collectors.toList());
-    }
-
-    return Stream.concat(
-            defaultIcons.values().stream()
-                .filter(
-                    icon ->
-                        Set.of(icon.getKeywords()).containsAll(iconOperationParams.getKeywords()))
-                .toList()
-                .stream(),
-            customIconStore.getIcons(iconOperationParams))
-        .collect(Collectors.toList());
+  public Set<CustomIcon> getCustomIcons(CustomIconOperationParams iconOperationParams) {
+    return customIconStore.getCustomIcons(iconOperationParams);
   }
 
   @Override
   @Transactional(readOnly = true)
-  public long count(IconOperationParams iconOperationParams) {
+  public long count(CustomIconOperationParams iconOperationParams) {
```

Comment:
Please avoid names like `iconOperationParams` that only repeat the type name. Long names like that add no new information to the code. Either use a short name like `params` or a name that adds new information, like `filters`, `criteria`, `restrictions` or alike.

Reply:
fixed